### PR TITLE
support profiling of modern cli applications

### PR DESCRIPTION
**Pull request by kiros32**

### CODE REVIEW

**Comment by aik099** [comment]
Please show some examples when profiling doesn't work.

**Comment by kiros32** [comment]
Sorry, I missed your message

Easiest example
I'm using Yii2 framework
When I run any cli command it looks like this:
`# php console/yii controller_name/action_name param1 param2`

So with current code
`$url   = isset($_SERVER['REQUEST_URI']) ? $_SERVER['REQUEST_URI'] : $_SERVER['PHP_SELF'];`
In $url variable I will have something like `/path/to/project/console/yii` and I don't see any parameters.

Also in my diff you can find the second place when current code does not even check if $_SERVER['REQUEST_URI'] exists
`$sql['c_url'] = $this->db->escape(_urlSimilartor($_SERVER['REQUEST_URI']));`

So that's why a made this merge request :)

File: xhprof_lib/utils/xhprof_runs.php [hunk 1]
```diff
@@ -442,11 +442,11 @@ public function save_run($xhprof_data, $type, $run_id = null, $xhprof_details =
 		}
 			
         
-	$url   = isset($_SERVER['REQUEST_URI']) ? $_SERVER['REQUEST_URI'] : $_SERVER['PHP_SELF'];
+	$url   = isset($_SERVER['REQUEST_URI']) ? $_SERVER["REQUEST_URI"] : implode(' ', $_SERVER['argv']);
```
Comment (aik099) [inline xhprof_lib/utils/xhprof_runs.php]: Please change to:

```php
$url   = PHP_SAPI === 'cli' ? implode(' ', $_SERVER['argv']) : $_SERVER['REQUEST_URI'];
```

---

Compared to your original change I've replaced `isset($_SERVER['REQUEST_URI'])` with `PHP_SAPI === 'cli'` to clearly indicate, that request url isn't set in CLI.

Comment (kiros32) [reply xhprof_lib/utils/xhprof_runs.php]: Ok, done

**Comment by aik099** [comment]
Merging, thanks @kiros32 .

**Pull request merged**